GVFS/FastFetch/NativeUnixMethods.cs

```diff
@@ -14,21 +14,35 @@ public class NativeUnixMethods
         public const int Create = 0x0200;
         public const int Truncate = 0x0400;
         private const int InvalidFileDescriptor = -1;
+        private const ushort SymLinkMode = 0xA000;
 
         public static unsafe void WriteFile(ITracer tracer, byte* originalData, long originalSize, string destination, ushort mode)
         {
             int fileDescriptor = InvalidFileDescriptor;
             try
             {
-                fileDescriptor = Open(destination, WriteOnly | Create | Truncate, mode);
-                if (fileDescriptor != InvalidFileDescriptor)
+                if (mode == SymLinkMode)
                 {
-                    IntPtr result = Write(fileDescriptor, originalData, (IntPtr)originalSize);
-                    if (result.ToInt32() == -1)
+                    string linkLocation = Marshal.PtrToStringUTF8(new IntPtr(originalData));

```

Comment:
Naming suggestion: `linkLocation` could be interpreted as the location at which the symlink file will be created.  Consider renaming it to something like `linkTarget`.